addons/product/models/product.py

```diff
@@ -108,6 +108,7 @@ class ProductProduct(models.Model):
         'Barcode', copy=False,
         help="International Article Number used for product identification.")
     product_template_attribute_value_ids = fields.Many2many('product.template.attribute.value', relation='product_variant_combination', string="Attribute Values", ondelete='restrict')
+    product_template_variant_value_ids = fields.Many2many('product.template.attribute.value', compute='_compute_variant_attributes')
```

Comment:
Why don't you use a domain, instead of a computed field? It will be faster and less code.